### ISSUE TITLE
Show classes results after each epochs

### ISSUE THREAD

**Issue by osamaalschame** [ultralytics/ultralytics#17862]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, 
Previously, using verbose argument will show the results for each class after each epoch. now, it does not work, how to show the classes results after each epoch.

regards,

**Comment by UltralyticsAssistant** [comment]
👋 Hello @osamaalschame, thank you for reaching out and for your interest in Ultralytics 🚀! 

We recommend checking out the [Docs](https://docs.ultralytics.com) for helpful tips, examples, and updates, including detailed guidance on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage.

If this is a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) and share any relevant training configurations, environment details, or logs? This will help us debug and understand the issue further.

If this is a ❓ Question regarding custom training, please provide as much information as possible, including your current training setup (e.g., commands used, model configuration, environment) and specific details about the issue you're encountering to allow for better assistance. Additionally, please ensure you are following the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Upgrade

Please make sure you’re using the latest version of the `ultralytics` package. You can upgrade in your environment with:

```bash
pip install -U ultralytics
```

This ensures you're running with the latest features and bug fixes. You can also check out our [requirements file](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) for a full list of dependencies.

## Environments

YOLO can be run in any of the following supported environments with dependencies preinstalled for you:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community and Support

Join the Ultralytics community where it suits you best:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧.
- Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com).
- Or engage with our [Subreddit](https://reddit.com/r/Ultralytics) to share ideas and get insights from fellow users.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If the badge above is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. These tests run daily and validate that all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) are functioning properly across supported environments.

This is an automated response to help guide you ⚡. An Ultralytics engineer will follow up as soon as possible to assist further! 😊

**Comment by Y-T-G** [comment]
It's no longer supported. It will only print all of them in the final validation.

**Comment by Y-T-G** [comment]
You can remove `not self.training` if you want to modify the code:

https://github.com/ultralytics/ultralytics/blob/c1554935e6568ec4b27fc0c25d7e36bb32aaa24c/ultralytics/models/yolo/detect/val.py#L197

**Comment by osamaalschame** [comment]
> You can remove `not self.training` if you want to modify the code to enable it:
> 
> https://github.com/ultralytics/ultralytics/blob/c1554935e6568ec4b27fc0c25d7e36bb32aaa24c/ultralytics/models/yolo/detect/val.py#L197

it does not work

**Comment by Y-T-G** [comment]
It works fine for me. You probably didn't apply the change correctly or not in the right file.

**Comment by Y-T-G** [comment]
```python
Logging results to /ultralytics/runs/detect/train
Starting training for 10 epochs...
Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       1/10       2.8G      1.158       1.23      1.214        127
                 Class     Images  Instances      Box(P          R
                   all        128        929      0.651      0.604      0.675      0.504
                person         61        254      0.783      0.713      0.787      0.541
               bicycle          3          6      0.628      0.295      0.453      0.288
                   car         12         46      0.612      0.196       0.26      0.173
            motorcycle          4          5      0.708          1      0.995      0.789
              airplane          5          6       0.84      0.833      0.972      0.857
```

**Comment by osamaalschame** [comment]
> ```python
> Logging results to /ultralytics/runs/detect/train
> Starting training for 10 epochs...
> Closing dataloader mosaic
> albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))
> 
>       Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
>        1/10       2.8G      1.158       1.23      1.214        127
>                  Class     Images  Instances      Box(P          R
>                    all        128        929      0.651      0.604      0.675      0.504
>                 person         61        254      0.783      0.713      0.787      0.541
>                bicycle          3          6      0.628      0.295      0.453      0.288
>                    car         12         46      0.612      0.196       0.26      0.173
>             motorcycle          4          5      0.708          1      0.995      0.789
>               airplane          5          6       0.84      0.833      0.972      0.857
> ```

what arguments you use it model.train() to show this

**Comment by Y-T-G** [comment]
`results = model.train(data="coco128.yaml", epochs=10)`

**Comment by Y-T-G** [comment]
@Fritskee Validation losses are only calculated during validation when training. During `model.val()`, it's not calculated.

**Comment by Fritskee** [comment]
> @Fritskee Validation losses are only calculated during validation when training. During `model.val()`, it's not calculated.

I'm aware, but just wondering how I can easily add different metrics than `'metrics/precision(B)', 'metrics/recall(B)', 'metrics/mAP50(B)', 'metrics/mAP50-95(B)'`. It seems like it's all quite "deep" in the codebase, but I figured there must be an easier way that I'm overlooking.

**Comment by glenn-jocher** [comment]
To add custom metrics, you'll need to modify the `validator` logic in the codebase. You can extend the `DetectionValidator` class or add your desired metrics directly in the validation loop. Unfortunately, there's no built-in simpler way to achieve this currently. For guidance, you can review the `ultralytics/engine/validator.py` file in the codebase.

**Comment by glenn-jocher** [comment]
Yes, you're correct. If you're working with an OBB model, you'd need to modify the `OBBValidator` class in `models/yolo/obb/val.py`. To add the AP50 for the first class, you can customize the `metrics` computation within the `OBBValidator` class. Specifically, look at how `OBBMetrics` is implemented and extend it to calculate and store the AP50 for the desired class. You can find the relevant code [here](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/models/yolo/obb/val.py).

**Comment by KennyTC** [comment]
> You can remove `not self.training` if you want to modify the code to enable it:
> 
> https://github.com/ultralytics/ultralytics/blob/c1554935e6568ec4b27fc0c25d7e36bb32aaa24c/ultralytics/models/yolo/detect/val.py#L197

It worked with coco dataset, but did not work with custom dataset. How can I make it work with custom dataset?
